RZSequenceController.h

```diff
@@ -0,0 +1,208 @@
+ *  @param fromVC             The current view controller in the sequence.
+ *  @param toVC               The view controller that will become the current view controller in the sequence when the transition is complete.
+ */
+- (UIPercentDrivenInteractiveTransition *)sequenceController:(RZSequenceController *)sequenceController transitionFromViewController:(UIViewController *)fromVC toViewController:(UIViewController *)toVC;
+
+@end
+
+/**
+ *  An RZSequenceControllerDelegate receives callbacks from an RZSequenceController related to transition events in the sequence.
+ */
+@protocol RZSequenceControllerDelegate <NSObject>
+
+@optional
+/**
+ *  Called by the sequence controller before a transition to determine whether to transition from its current view controller to the view controller at the given index.
+ *
+ *  @return YES if the transition should take place, and NO otherwise. If NO, the scroll view will bounce back instead of scrolling.
+ */
+- (BOOL)sequenceController:(RZSequenceController *)sequenceController shouldTransitionToIndex:(NSUInteger)idx;
+
+/**
+ *  Called when a transition is about to take place from the current view controller to the given view controller.
+ *  At this point the currentChild and currentIndex properties of the sequence controller have not been changed.
+ *
+ *  @param sequenceController   The sequence controller that will perform a transition.
+ *  @param child                The child that will be transitioned to.
+ *  @param idx                  The index that will be transitioned to.
+ *  @param animated             Whether the transition will be animated.
+ */
+- (void)sequenceController:(RZSequenceController *)sequenceController willTransitionToChild:(UIViewController<RZSequenceChildViewController> *)child atIndex:(NSUInteger)idx animated:(BOOL)animated;
+
+/**
+ *  Called after a transition occurs from one view controller to another.
+ *  At this point the currentChild and currentIndex properties of the sequence controller have been updated.
+ *
+ *  @param sequenceController   The sequence controller that performed a transition.
+ *  @param child                The child that was transitioned away from. This view controller may no longer be in the heirarchy.
+ *  @param animated             Whether the transition was animated.
+ */
+- (void)sequenceController:(RZSequenceController *)sequenceController didTransitionFromChild:(UIViewController<RZSequenceChildViewController> *)child animated:(BOOL)animated;
+
+@end
+
+/**
+ *  A view controller container that arranges child view controller in either a horizontal or vertical scrolling flow.
+ *  The container managers transitions between children using UIPercentDrivenInteractiveTransitions driven by scroll events.
+ */
+@interface RZSequenceController : UIViewController
+
+/**
+ *  The direction of the sequence flow, either RZSequenceDirectionHorizontal or RZSequenceDirectionVertical.
+ */
+@property (nonatomic, readonly) RZSequenceDirection direction;
```

Comment:
Style guide doesn't say this is required. I prefer to leave it out, since it is always redundant for primitives. I never use strength specifiers except for `strong`, `weak`, and `copy`. But since it’s not in the style guide, it’s personal preference.